### ISSUE TITLE
eval_Hpatches

### ISSUE THREAD

**Issue by wjk-imagematching** [chicleee/EDM#11]
Hi! Could you share the exact script or hyperparameters used for the HPatches evaluation so we can reproduce the paper numbers?

**Comment by wjk-imagematching** [comment]
我的参数配置是粗thr0.3，config.EDM.COARSE.TOPK = 1000，ransac阈值3，图像都padding为480*480的正方形，指标如下：
==== Homography Estimation ====
Hest solver=cv est_failed=1 ransac_thres=3.0 inlier_rate=0.95
Hest Correct: a=[0.65 0.88 0.92 0.95]
i=[0.83 0.99 0.99 1.  ]
v=[0.48 0.78 0.86 0.9 ]
Hest AUC: a=[39.96 67.08 76.56 85.32]
i=[54.26 81.44 88.41 93.97]
v=[26.85 53.9  65.66 77.43]

**Comment by chicleee** [comment]
We follow this repo [image-matching-toolbox](https://github.com/TruongKhang/image-matching-toolbox) to evaluate HPatches. 
The parameters and results are shown in the following figure.
![Image](https://github.com/user-attachments/assets/16469e4d-7175-4f26-85d0-39facff2c5f8)

**Comment by wjk-imagematching** [comment]
非常感谢你的回复，我使用了您提供的评估链接重新进行了测试，发现指标如下：
>>>> Method=EDM Default config: {'class': 'EDM', 'ckpt': '/home/wujunke/EDM-main/weights/edm_outdoor.ckpt', 'match_threshold': 0.3, 'imsize': 480, 'no_match_upscale': True, 'dim_resized': 'min'} Thres: [0.3]
Initialize EDM_edm_outdoor
Matching thres: 0.3  Save to: None

>>>>Eval hpatches: task=homography method=EDM_edm_outdoor scale_H=True rthres=6 thres=[1, 3, 5, 10] 
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 108/108 [00:13<00:00,  7.79it/s]
>>Finished, pairs=290 match_failed=250 matches=522.6 match_time=0.02s
==== Homography Estimation ====
Hest solver=cv est_failed=250 ransac_thres=6 inlier_rate=0.53
Hest Correct: a=[37.78 49.26 51.3  52.22]
i=[42.69 48.08 48.08 48.08]
v=[33.21 50.36 54.29 56.07]
Hest AUC: a=[24.32 38.38 43.24 47.65]
i=[32.6  42.41 44.68 46.38]
v=[16.81 34.73 41.96 48.86]

不知道是哪里出现了问题。但是值得注意的是，您在EDM_config中的一些其他配置如：max_n_matches,match_border_rm以及match_topk我不清楚如何使用，这可能导致了效果不好。

**Comment by wjk-imagematching** [comment]
如果可以，能不能好心给下edm模块的代码（qq邮箱：[1061427975@qq.com](mailto:1061427975@qq.com)），或者指教一下，我将不胜感激，万分感谢！
我的edm模块代码如下：


``import sys
import torch
import torch.nn.functional as F
import numpy as np
from pathlib import Path
from argparse import Namespace
from third_party.edm.src.edm import EDM as EDM_
from third_party.edm.src.config.default import get_cfg_defaults
from third_party.edm.src.utils.misc import lower_config
from .base import Matching
from immatch.utils.data_io import load_gray_scale_tensor_cv
import cv2

class EDM(Matching):
    def __init__(self, args):
        super().__init__()
        if type(args) == dict:
            args = Namespace(**args)

        self.imsize = args.imsize
        self.resize_type = max if args.dim_resized == 'max' else min
        self.match_threshold = args.match_threshold
        self.no_match_upscale = args.no_match_upscale
        # self.eval_coarse = args.eval_coarse

        # --- 3. 配置构造：严格对齐 480x640 逻辑 ---
        config = get_cfg_defaults()
        config.defrost()


        config.EDM.NECK.NPE = [832, 832, 1216, 1216]

        config.EDM.COARSE.TOPK = 1000
        config.EDM.COARSE.MCONF_THR = self.match_threshold

        config.freeze()

        # 4. 初始化模型
        _config = lower_config(config)
        self.model = EDM_(config=_config["edm"])

        # 5. 加载权重
        ckpt_dict = torch.load(args.ckpt, map_location=self.device)
        state_dict = ckpt_dict.get("state_dict", ckpt_dict)
        new_state_dict = {k.replace("matcher.", ""): v for k, v in state_dict.items()}
        self.model.load_state_dict(new_state_dict, strict=False)
        self.model = self.model.eval().to(self.device)
        # # 初始全局 RoPE 更新：强制使用 480x640 的频率映射
        # self.setup_rope_static()
        self.name = f"EDM_{args.ckpt.split('/')[-1].split('.')[0]}"
        print(f"Initialize {self.name}")


    def load_im(self, im_path):
        return load_gray_scale_tensor_cv(
            im_path, self.device, imsize=self.imsize, dfactor=8, scale_type=self.resize_type
        )

    def match_inputs_(self, gray1, gray2):
        batch = {'image0': gray1, 'image1': gray2}
        self.model(batch)
        kpts1 = batch['mkpts0_f'].cpu().numpy()
        kpts2 = batch['mkpts1_f'].cpu().numpy()
        scores = batch['mconf'].cpu().numpy()
        matches = np.concatenate([kpts1, kpts2], axis=1)
        return matches, kpts1, kpts2, scores

    def match_pairs(self, im1_path, im2_path):
        gray1, sc1 = self.load_im(im1_path)
        gray2, sc2 = self.load_im(im2_path)

        upscale = np.array([sc1 + sc2])
        matches, kpts1, kpts2, scores = self.match_inputs_(gray1, gray2)

        if self.no_match_upscale:
            return matches, kpts1, kpts2, scores, upscale.squeeze(0)

        # Upscale matches &  kpts
        matches = upscale * matches
        kpts1 = sc1 * kpts1
        kpts2 = sc2 * kpts2
        return matches, kpts1, kpts2, scores``

**Comment by chicleee** [comment]
完整的评估代码已邮件发送~请查收

**Comment by wjk-imagematching** [comment]
非常感谢！
